app/src/main/java/ch/epfl/cs311/wanderwave/ui/components/player/ExclusivePlayer.kt

```diff
@@ -230,16 +227,16 @@ fun PlayPauseButton(viewModel: TrackListViewModel, uiState: TrackListViewModel.U
 }
 
 @Composable
-fun RepeatButton(viewModel: TrackListViewModel, uiState: TrackListViewModel.UiState) {
-  IconButton(onClick = { viewModel.toggleLoop() }, modifier = Modifier.testTag("toggleRepeat")) {
-    when (uiState.loopMode) {
-      LoopMode.NONE ->
+fun RepeatButton(viewModel: PlayerViewModel, uiState: PlayerViewModel.UiState) {
+  IconButton(onClick = { viewModel.toggleRepeat() }, modifier = Modifier.testTag("toggleRepeat")) {
+    when (uiState.repeatMode) {
+      false ->
```

Comment:
Was there any particular reason for removing the ability to loop over one single track ? It's a feature that is available in most (if not all) media players so ideally we should keep it.

Reply:
I can do it if it's too much of a hassle, let me know

Reply:
Yeah I did, but now the tests are acting weird. There's not really any useful output on the CI, but when I run the tests locally, a seemingly random one fails each time, also with minimal output... Have you experienced this before?
I just tried running 'All tests' locally, and the first time, makeApiRequest_ShouldReturnExpectedResult_OnSuccessfulApiCall() failed, the second time it was the e2e-test (timeout I think) and the third time spotifyLoginRunsIntent(). Very strange

Reply:
Edit: I have the same problem on the main branch as well, locally, so maybe it's unrelated.
Are you able to run the tests on your device consistently? Does anything fail there?